images/bootstrap/Dockerfile

```diff
@@ -60,20 +61,33 @@ RUN dnf install -y \
     python3-jinja2 &&\
   dnf -y clean all
 
-# Install gcloud
-ENV PATH=/google-cloud-sdk/bin:/workspace:${PATH} \
-    CLOUDSDK_CORE_DISABLE_PROMPTS=1
-
-RUN wget -q https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz && \
-    tar xzf google-cloud-sdk.tar.gz -C / && \
-    rm google-cloud-sdk.tar.gz && \
-    /google-cloud-sdk/install.sh \
-        --disable-installation-options \
-        --bash-completion=false \
-        --path-update=false \
-        --usage-reporting=false && \
-    gcloud components install alpha beta kubectl && \
-    gcloud info | tee /workspace/gcloud-info.txt
+
+RUN if test "${ARCH}" != s390x; then \
+        export PATH=/google-cloud-sdk/bin:/workspace:${PATH} && \
+        export CLOUDSDK_CORE_DISABLE_PROMPTS=1 && \
+        wget -q https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz && \
+        tar xzf google-cloud-sdk.tar.gz -C / && \
+        rm google-cloud-sdk.tar.gz && \
+        /google-cloud-sdk/install.sh \
+            --disable-installation-options \
+            --bash-completion=false \
+            --path-update=false \
+            --usage-reporting=false && \
+        gcloud components install alpha beta kubectl && \
+        gcloud info | tee /workspace/gcloud-info.txt; \
+        # Cache the most commonly used bazel versions in the container
+        curl -Lo ./bazelisk https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-${ARCH} && \
+        chmod +x ./bazelisk && mv ./bazelisk /usr/local/bin/bazelisk && \
+        cd /usr/local/bin && ln -s bazelisk bazel; \
+
+        # Cache the most commonly used bazel versions inside the image
+        # and remove resulting cache directories to save a few hundred MB
+        USE_BAZEL_VERSION=4.1.0 bazel version && \
+        USE_BAZEL_VERSION=4.2.1 bazel version && \
+        USE_BAZEL_VERSION=5.3.1 bazel version && \
+        USE_BAZEL_VERSION=5.4.1 bazel version && \
+        rm -rf /root/.cache/bazel; \
+        fi
```

Comment:
nit: could we remove one indentation step for the `fi`, so it becomes more obvious? I was looking to find it in the previous lines...